internals/overlord/checkstate/handlers.go

```diff
@@ -94,17 +101,36 @@ func (m *CheckManager) doPerformCheck(task *state.Task, tomb *tombpkg.Tomb) erro
 				m.state.Unlock()
 			}
 		}
-		return false, nil
+		return false, err
```

Comment:
I think it'd be simpler/clearer to change the `if err != nil` / `else` to just an `if` with an early `return false, err`. And then remove and un-indent the `else` block, with `return false, nil` at the end. This would make it the same structure as `recoverCheck`.